### PR TITLE
#50: add the ability to create sitemap index

### PR DESCRIPTION
**Pull request by ManuDoni**
#50: add SitemapIndex class to create sitemap index by passing urls of already created sitemaps.
Extra: add test for Sitemap\Tags\Url `setUrl `method, now code coverage is 100%

### CODE REVIEW

File: src/SitemapIndex.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        return new static();
+    }
+
+    /**
+     * @param string|\Spatie\Sitemap\Tags\Tag $tag
+     *
+     * @return $this
+     */
+    public function add($tag)
```
Comment (freekmurze) [inline src/SitemapIndex.php]: Wouldn't `$url` be a better name than `$tag`?

File: src/Tags/Sitemap.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        return $segments;
+    }
+
+    /**
+     * @param int $index
+     *
+     * @return string|null
+     */
+    public function segment(int $index)
```
Comment (freekmurze) [inline src/Tags/Sitemap.php]: Is this function needed?

File: src/SitemapIndex.php (hunk 1 above)
Comment (ManuDoni) [inline src/SitemapIndex.php]: I think you are right, I called it `$tag` for consistency with the `Sitemap\Sitemap` class

File: src/Tags/Sitemap.php (hunk 2 above)
Comment (ManuDoni) [inline src/Tags/Sitemap.php]: Same as `segments` method, it can be dropped.

**Comment by ManuDoni** [comment]
Tell me if you think I can drop `segment` and `segments` methods, I'll make a commit for it

**Comment by freekmurze** [comment]
Yup, drop 'm

**Comment by ManuDoni** [comment]
Could you rerun Scrutinizer analysis? It seems it's failed for an internal error.

**Comment by freekmurze** [comment]
Restarted the analysis

**Comment by ManuDoni** [comment]
The status is still "Errored", I thought it could refresh the status of the analysis. I don't know if it deals or not with my code.
Thanks anyway!
When you'll have time could you check and merge this PR?

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**